include/aspect/material_model/interface.h

```diff
@@ -409,6 +409,12 @@ namespace aspect
          */
         std::vector<SymmetricTensor<2,dim>> strain_rate;
 
+        /**
+         * The products of the determinant of Jacobian and the quadrature weight
+         * at the points given in the #position vector.
+         */
+        std::vector<double> JxW;
```

Comment:
I think this is perhaps the wrong approach. It assumes that the material model inputs are only used for quadrature, but this object is also used for evaluation at arbitrary points or sets of points for which there is no sense of integrating anything. I wonder how that could be generalized.